webview-ui/src/components/common/MermaidBlock.tsx

```diff
@@ -139,12 +143,90 @@ export default function MermaidBlock({ code }: MermaidBlockProps) {
 		}
 	}
 
+	/**
+	 * Copy the mermaid code to clipboard for easier fixing
+	 */
+	const handleCopyCode = () => {
+		navigator.clipboard
+			.writeText(code)
+			.then(() => {
+				setShowCopySuccess(true)
+				setTimeout(() => {
+					setShowCopySuccess(false)
+				}, 1000)
+			})
+			.catch((err) => {
+				console.error("Failed to copy code:", err)
+			})
+	}
+
 	return (
 		<MermaidBlockContainer>
-			{isLoading && <LoadingMessage>Generating mermaid diagram...</LoadingMessage>}
-
-			{/* The container for the final <svg> or raw code. */}
-			<SvgContainer onClick={handleClick} ref={containerRef} $isLoading={isLoading} />
+			{isLoading && <LoadingMessage>{t("common:mermaid.loading")}</LoadingMessage>}
+
+			{error ? (
+				<div style={{ marginTop: "0px", overflow: "hidden", marginBottom: "8px" }}>
```

Comment:
Should we run this past the "Design Engineer" mode? Seems like a lot of could be done with Tailwind.

Reply:
I did use the Design Engineer mode fwiw... maybe because there's so much surrounding styled components it decided to match?

Reply:
I didn't explicitly ask for Tailwind - I just asked it to build an error display like we have for unsuccessful diffs.

Reply:
(in case it was lost below): Use `<CodeAccordian>` which is what all the other tool chevron-expander things use (it embeds `<CodeBlock>` and Shiki supports mermaid).

Reply:
I just updated to use `<CodeBlock language="mermaid" source={code}>` for consistency, though I'm not sure how well the syntax highlighting will work on the invalid mermaid.